homework-8/script.js

```diff
@@ -0,0 +1,93 @@
+// From the base FC Human, you need to implement inheritance for FC Teacher and FC Student
+
+// 1. Create FC Human, which accepts an object as arguments and creates properties name, surname and age. FC Human also contains methods:
+// getFullName() - returns the full name string formed from name, surname,
+// setFullName(fullName) - splits the string into name, surname
+// The surname, name and age properties must be present in all FCs. Eliminate duplication with context substitution.
+
+// 2. Create FC Student, which I can use as follows:
+
+// let s = new Student(info) // where info.marks = [10,9,8,1,10] and all other properties name, age ...
+// Every student should have such methods
+// getAverageMark() - returns the average mark
+// getMinMark() - returns the smallest mark
+// getMaxMark() - returns the maximum mark
+
+// 3. Create a Teacher FC that accepts an object as an argument and creates a group property (an array of at least 5 students created using the Student FC) and contains methods:
+// - getListOfNamesByAverageMark() - returns an array of student names sorted by the highest average mark.
+// - getStudentByName(name) - get one student object by name.
+// - removeStudentByName(name) - remove the student object found by name.
+// - updateStudentByName(new Student(...), name) - find the student object by name and replace it with student (a new instance of FC Student)
+
+// (FC - constructor function)
+
+// FC Human constructor function
+function Human(obj) {
+  this.name = obj.name;
+  this.surname = obj.surname;
+  this.age = obj.age;
+}
+
+Human.prototype.getFullName = function () {
+  return this.name + " " + this.surname;
+};
+
+Human.prototype.setFullName = function (fullName) {
+  const fullNameArr = fullName.split(" ");
+  this.name = fullNameArr[0];
+  this.surname = fullNameArr[1];
+};
+
+// Student constructor function
+function Student(obj) {
+  Human.call(this, obj);
+  this.marks = obj.marks;
+}
+
+Student.prototype = Object.create(Human.prototype);
+Student.prototype.constructor = Student;
+
+Student.prototype.getAverageMark = function () {
+  return this.marks.reduce((acc, val) => acc + val, 0) / this.marks.length;
```

Comment:
in this case you can use a, b naming convention, and don't pass acc parameter
```suggestion
  return this.marks.reduce((a, b) => a + b) / this.marks.length;
```